zpeclient.go

```diff
@@ -300,3 +334,48 @@ func (c *Cache) authorize(principal string, check AthenzAccessCheck) (bool, erro
 	}
 	return false, nil
 }
+
+// parseUpdateTime - read the last update time as string from configmap, store in the cache
+func (c *Cache) parseUpdateTime(configmap interface{}) error {
+	var lastUpdateTime string
+
+	obj, ok := configmap.(*corev1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("Unable to cast interface to config map object")
+	}
+
+	// check if lastUpdateKeyField exists in the configmap
+	if _, ok = obj.Data[lastUpdateKeyField]; ok {
+		lastUpdateTime = strings.ReplaceAll(obj.Data[lastUpdateKeyField], `"`, "")
+	} else {
+		return fmt.Errorf("configmap format is wrong, it doesn't have a field called: %v", lastUpdateKeyField)
+	}
+	var err error
+
+	// update last athenz contact time in the cache object
+	c.lock.Lock()
+	c.lastUpdate, err = time.Parse(time.RFC3339Nano, lastUpdateTime)
+	c.lock.Unlock()
```

Comment:
As discussed, let's make this a single function which sets both lastUpdate and cacheStatus from both the cron and the event handler. Something along the lines of:

```
func updateCacheStatus(timestamp string) {

	lock()
	defer unlock()

	if timestamp != "" {
		c.lastUpdateTime = timestamp
	}

	diff := t.sub(c.lastUpdateTime)
	if diff < c.maxcontacttime {
		c.cacheStatus = true
	} else {
		c.cacheStatus = false
	}
}
```